src/index.tsx

```diff
@@ -21,25 +21,17 @@ interface WorkerLoad<Data> {
   /** The worker's returned value stored in a state */
   data: Data;
 
-  /** An optional object that contains any thrown errors, if any, and a retry function */
-  error?: RetryWorkerError;
+  /** The error thrown by the worker's call */
+  error?: Error;
 
   /** Sets [[isLoading]] state manually */
   setIsLoading: (_: boolean) => void;
 
   /** Sets [[error]] state manually */
   setError: (_: Error | undefined) => void;
-}
-
-/**
- * Encapsulates [[useWorkerLoad]]'s errors and retry function
- */
-interface RetryWorkerError {
-  /** The error thrown by the worker's call */
-  value: Error;
 
   /** Calls the worker again */
-  retry: () => Promise<void>;
+  callback: () => Promise<void>;
```

Comment:
What do you think about adding an extra comment explaining that this callback is never updated?
It might be useful in case the user tries to use a dependence inside the `worker` like the following:

```tsx
const Home = () => {
  const [anyOtherLocalDataOrState, setLocalData] = useState('data initial value');
  const { isLoading, data, callback } = useWorkerLoad(async () => {
    const dataFromApi = await getDataFromApi();
    if (dataFromApi !== anyOtherLocalDataOrState) {
      // do something
    } else {
      // do something else
    }
  });
  // the rest...
};```

Reply:
We can also rename the `useWorkerLoad` argument from `worker` to `initialWorker` or something else that determines it won't accept a second worker or that only the first instance of the function will be executed.

Reply:
That is a good catch. I didn't think of that shortcoming. Also, it might make sense to rename it. Basically, the concept of this hook is the following:
1. It's a worker
2. It executes the callback automatically when the component is mounted.
3. It stores the value returned by the callback in a state.

What would be a good name for this hook? "Initial worker state"? "Auto worker state"? "Worker storage"?

With that in mind, does it still make sense to execute the worker on "didLoad"? We know for a fact it's a recurring pattern, but it's just one line of code.

Reply:
What about `useMountWorker`?

Reply:
That might be a good idea, but how do explicitly convey it stores the return value in a state?

Reply:
I'm not sure it should convey it stores the value in the state. [`useNetInfo`](https://github.com/react-native-community/react-native-netinfo/blob/master/src/index.ts#L105-L110) already do that and we just take it for granted.